src/main/java/seedu/address/logic/parser/RemoveTagCommandParser.java

```diff
@@ -0,0 +1,34 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.logic.commands.RemoveTagCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Parses input argument and creates a new RemoveTagCommand object.
+ */
+public class RemoveTagCommandParser implements Parser<RemoveTagCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the RemoveTagCommand
+     * and returns a RemoveTagCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public RemoveTagCommand parse(String args) throws ParseException {
+        try {
+            String trimmedArgs = args.trim();
+            System.out.println(Tag.isValidTagName(trimmedArgs));
+            if (Tag.isValidTagName(trimmedArgs)) {
+                return new RemoveTagCommand(trimmedArgs);
+            } else {
+                throw new IllegalValueException("Invalid input value for tag!");
```

Comment:
Why throw an `IllegalValueException` just to catch it immediately after? You can just throw the `ParseException` here directly.